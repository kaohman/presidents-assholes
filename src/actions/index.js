export const setPresidents = (presidents) => ({
  type: 'SET_PRESIDENTS',
  presidents
});

export const setLoading = (loading) => ({
  type: 'SET_LOADING',
  loading
});

export const setError = (error) => ({
  type: 'SET_ERROR',
  error
});

