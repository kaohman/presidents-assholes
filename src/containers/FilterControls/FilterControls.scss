.filter-container {
  width: 155px;
  margin: 10px auto;
}