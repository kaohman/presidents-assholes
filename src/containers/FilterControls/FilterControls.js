import React from 'react';
import { connect } from 'react-redux'; 
import { setParty } from '../../actions';

export const FilterControls = () => {

  return (
    <div>
      <select>
        <option>

        </option>
      </select>
    </div>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  setParty: (party) => dispatch(setParty(party)),
});

export default connect(null, mapDispatchToProps)(FilterControls);