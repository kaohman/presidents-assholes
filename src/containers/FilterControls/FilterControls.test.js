import { FilterControls, mapDispatchToProps } from './FilterControls';
import { setParty } from '../../actions';
import { shallow } from 'enzyme';
import React from 'react';

describe('FilterControls', () => {
  let wrapper;
  let setPartyMock;

  beforeEach(() => {
    setPartyMock = jest.fn();
    wrapper = shallow(
      <FilterControls
        setParty={setPartyMock}
      />
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('updateParty', () => {
    it('should update the party when a new option is selected', () => {
      const mockEvent = { target: { value: 'Whig' } };
      wrapper.find('select').simulate('change', mockEvent);
      expect(setPartyMock).toHaveBeenCalledWith('Whig');
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with setParty is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setParty();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setParty();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});