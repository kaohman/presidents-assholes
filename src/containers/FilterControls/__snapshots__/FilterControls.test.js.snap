// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterControls should match the snapshot 1`] = `
<div
  className="filter-container"
>
  <select
    onChange={[Function]}
  >
    <option
      default={true}
      value=""
    >
      All
    </option>
    <option
      value="No Party"
    >
      No Party
    </option>
    <option
      value="Federalist"
    >
      Federalist
    </option>
    <option
      value="Democratic-Republican"
    >
      Democratic-Republican
    </option>
    <option
      value="Whig"
    >
      Whig
    </option>
    <option
      value="Democratic"
    >
      Democratic
    </option>
    <option
      value="Republican"
    >
      Republican
    </option>
  </select>
</div>
`;
